plugin/src/main/java/org/opensearch/ml/task/MLTaskRunner.java

```diff
@@ -117,6 +118,19 @@ public void dispatchTask(
                 // Execute ML task remotely
                 log.debug("Execute ML request {} remotely on node {}", request.getRequestID(), nodeId);
                 request.setDispatchTask(false);
+                if (isStreamingRequest(request)) {
+                    log.debug("Using streaming transport for request {}", request.getRequestID());
+                    transportService
+                        .sendRequest(
+                            node,
+                            getTransportStreamActionName(),
+                            request,
+                            TransportRequestOptions.builder().withType(TransportRequestOptions.Type.STREAM).build(),
+                            getResponseStreamHandler(request)
+                        );
+                } else {
+                    transportService.sendRequest(node, getTransportActionName(), request, getResponseHandler(listener));
+                }
                 transportService.sendRequest(node, getTransportActionName(), request, getResponseHandler(listener));
```

Comment:
is this sendRequest duplicated in this logic? it looks like jumping to else condition , it will call two times